### DISCUSSION TITLE
覆盖主题文件后提示 warn no layout

### DISCUSSION THREAD

**Discussion by Nofated095** [ppoffice/hexo-theme-icarus#1117] [Q&A] [answered]
如题所示，我用 package.json 安装 hexo-theme-icarus，我希望对某部分主题的 jsx 文件做自定义修改，[#892](https://github.com/ppoffice/hexo-theme-icarus/discussions/892#discussioncomment-1032657) 中提出了通过创建 jsx 文件来覆盖主题中的 jsx，于是我尝试创建了 [layout.jsx](https://github.com/Nofated095/blog/commit/134888594405095a55fbef893bae9a686677fdf4)
![39859E2F-6749-47B0-9392-34B9D22ECD2A](https://user-images.githubusercontent.com/49985975/190857956-35419b99-1329-4345-8e86-a6c3a8386cf4.png)
但在部署时出现了 warn，站点构建成功但完全无法显示。

**Comment by ppoffice** [comment, marked as answer]
`layout.jsx`不可以用覆盖的方式来修改。能用覆盖方式修改的只有`comment`, `donate`, `plugin`, `search`, `share`, `widget`文件夹下面的jsx文件。

**Comment by Nofated095** [comment]
@ppoffice [waline.jsx](https://github.com/Nofated095/blog/blob/0cf7699ad375a432496d7867a82b6e4352909d35/themes/icarus/layout/comment/waline.jsx#L6) 这一行导致了无法正确运用文件，有什么方法可以让他从 npm 包里读取文件吗
![BDFD9090-8B23-49AA-9568-999D753F5E5F](https://user-images.githubusercontent.com/49985975/190881295-6ad81547-6cb3-468f-81a0-43687aa7df39.png)

**Reply by ppoffice** [reply]
https://ppoffice.github.io/hexo-theme-icarus/uncategorized/%E5%B8%B8%E8%A7%81%E9%97%AE%E9%A2%98/#%E5%B8%83%E5%B1%80

**Comment by Nofated095** [comment]
![0AB1B2FE-3D88-4A1C-B4E0-BEC7A8854E38](https://user-images.githubusercontent.com/49985975/190883657-a4e2f3c4-b146-4bc0-ae64-886a25bf37c3.png)
修正了引用路径，但仍然提示 no layout

[waline.jsx](https://github.com/Nofated095/blog/blob/dcd16e9846985acbd352ca73900adbbd0e48bd03/themes/icarus/layout/comment/waline.jsx)

**Reply by ppoffice** [reply]
你需要通过源码安装而不是通过npm安装Icarus。
